enable Travis CI

To improve the workflow, I think it is a good improvement to enable travis at this repo.

we can use there `cppcheck` and build all the files with `gcc`, for example

```
$ gcc -Wall -Wextra `pkg-config --cflags glib-2.0 gio-2.0 gtk+-3.0` eggdesktopfile.c `pkg-config --libs glib-2.0 gio-2.0 gtk+-3.0`
eggdesktopfile.c: In function ‘egg_desktop_file_launchv’:
eggdesktopfile.c:1305:33: warning: cast between incompatible function types from ‘void (*)(void *)’ to ‘void (*)(void *, void *)’ [-Wcast-function-type]
 1305 |       g_ptr_array_foreach (env, (GFunc)g_free, NULL);
      |                                 ^
```

--- a/.build.yml
+++ b/.build.yml
@@ -1,0 +1,132 @@
+##########################################################
+# THE FOLLOWING LINES IS USED BY docker-build
+##########################################################
+requires:
+  archlinux:
+    # Useful URL: https://git.archlinux.org/svntogit/community.git/tree/engrampa
+    - autoconf-archive
+    - caja
+    - clang
+    - file
+    - gcc
+    - git
+    - gtk3
+    - json-glib
+    - make
+    - mate-common
+    - which
+    - yelp-tools
+
+  debian:
+    # Useful URL: https://github.com/mate-desktop/debian-packages
+    # Useful URL: https://salsa.debian.org/debian-mate-team/engrampa
+    - autoconf-archive
+    - autopoint
+    - clang
+    - clang-tools
+    - cppcheck
+    - gettext
+    - gcc
+    - git
+    - libcaja-extension-dev
+    - libglib2.0-dev
+    - libgtk-3-dev
+    - libjson-glib-dev
+    - libmagic-dev
+    - make
+    - mate-common
+    - yelp-tools
+
+  fedora:
+    # Useful URL: https://src.fedoraproject.org/cgit/rpms/engrampa.git
+    - autoconf-archive
+    - caja-devel
+    - clang
+    - clang-analyzer
+    - desktop-file-utils
+    - file-devel
+    - gcc
+    - git
+    - gtk3-devel
+    - json-glib-devel
+    - libSM-devel
+    - make
+    - mate-common
+    - redhat-rpm-config
+
+  ubuntu:
+    - autoconf-archive
+    - autopoint
+    - clang
+    - clang-tools
+    - gcc
+    - gettext
+    - git
+    - libcaja-extension-dev
+    - libglib2.0-dev
+    - libgtk-3-dev
+    - libjson-glib-dev
+    - libmagic-dev
+    - make
+    - mate-common
+    - yelp-tools
+
+variables:
+  - 'CHECKERS="
+    -enable-checker deadcode.DeadStores
+    -enable-checker alpha.deadcode.UnreachableCode
+    -enable-checker alpha.core.CastSize
+    -enable-checker alpha.core.CastToStruct
+    -enable-checker alpha.core.IdenticalExpr
+    -enable-checker alpha.core.SizeofPtr
+    -enable-checker alpha.security.ArrayBoundV2
+    -enable-checker alpha.security.MallocOverflow
+    -enable-checker alpha.security.ReturnPtrRange
+    -enable-checker alpha.unix.SimpleStream
+    -enable-checker alpha.unix.cstring.BufferOverlap
+    -enable-checker alpha.unix.cstring.NotNullTerminated
+    -enable-checker alpha.unix.cstring.OutOfBounds
+    -enable-checker alpha.core.FixedAddr
+    -enable-checker security.insecureAPI.strcpy"'
+
+before_scripts:
+  - cd ${START_DIR}
+  - '[ -f mate-common-1.24.1.tar.xz ] || curl -Ls -o mate-common-1.24.1.tar.xz https://github.com/mate-desktop/mate-common/releases/download/v1.24.1/mate-common-1.24.1.tar.xz'
+  - tar xf mate-common-1.24.1.tar.xz
+  - cd mate-common-1.24.1
+  - if [ ${DISTRO_NAME} == "debian" -o ${DISTRO_NAME} == "ubuntu" ];then
+  -     ./configure --prefix=/usr --libdir=/usr/lib/x86_64-linux-gnu --libexecdir=/usr/lib/x86_64-linux-gnu
+  - else
+  -     ./configure --prefix=/usr
+  - fi
+  - make
+  - make install
+
+build_scripts:
+  - cd ..
+  - git clone --depth 1 https://github.com/mate-desktop/engrampa.git
+  - cd engrampa
+  - cp -dpR ../rootdir/ .
+  - rm -R mate-submodules
+  - mv rootdir mate-submodules
+
+  - if [ ${DISTRO_NAME} == "debian" ];then
+  -     export CFLAGS+=" -Wsign-compare -Wunused-parameter"
+  -     cppcheck --enable=warning,style,performance,portability,information,missingInclude .
+  - fi
+
+  - NOCONFIGURE=1 ./autogen.sh
+  - scan-build $CHECKERS ./configure --enable-compile-warnings=maximum
+  - if [ $CPU_COUNT -gt 1 ]; then
+  -     if [ ${DISTRO_NAME} == "debian" ];then
+  -         scan-build $CHECKERS --keep-cc --use-cc=clang --use-c++=clang++ -o html-report make -j $(( $CPU_COUNT + 1 ))
+  -         make clean
+  -     fi
+  -     scan-build $CHECKERS --keep-cc -o html-report make -j $(( $CPU_COUNT + 1 ))
+  - else
+  -     if [ ${DISTRO_NAME} == "debian" ];then
+  -         scan-build $CHECKERS --keep-cc --use-cc=clang --use-c++=clang++ -o html-report make
+  -         make clean
+  -     fi
+  -     scan-build $CHECKERS --keep-cc -o html-report make
+  - fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,38 @@
+# vim: set ts=2 sts=2 sw=2 expandtab :
+dist: bionic
+language: shell
+os: linux
+services:
+  - docker
+addons:
+  apt:
+    packages:
+      - python3-github
+
+before_install:
+  - curl -Ls -o docker-build https://github.com/mate-desktop/mate-dev-scripts/raw/master/travis/docker-build
+  - chmod +x docker-build
+
+install:
+  - ./docker-build --name ${DISTRO} --config .build.yml --install
+
+script:
+  - ./docker-build --name ${DISTRO} --verbose --config .build.yml --build scripts
+
+notifications:
+  irc:
+    if: branch = master AND
+        repo = "mate-desktop/mate-submodules"
+    channels:
+      - "irc.freenode.org#mate-dev"
+    template:
+      - "[%{repository_name}] %{author}: %{commit_subject}"
+      - "[%{branch}] %{commit} %{message} %{build_url}"
+    on_success: never
+    on_failure: always
+
+env:
+  - DISTRO="archlinux:latest"
+  - DISTRO="debian:testing"
+  - DISTRO="fedora:latest"
+  - DISTRO="ubuntu:devel"

